dask/array/_shuffle.py

```diff
@@ -79,7 +87,44 @@ def shuffle(x, indexer: list[list[int]], axis):
     return Array(graph, out_name, chunks, meta=x)
 
 
-def _shuffle(chunks, indexer, axis, in_name, out_name, token):
+def _resize_other_dimensions(x: Array, longest_group, axis, chunks):
+    assert chunks == "auto", "Only auto is supported for now"
+    chunksize_tolerance = config.get("array.shuffle.chunksize-tolerance")
+
+    if longest_group <= max(x.chunks[axis]) * chunksize_tolerance:
+        # We are staying below our threshold, so don't rechunk
+        return x
+
+    max_group = reduce(mul, map(max, x.chunks))
+    new_max_group = max_group / max(x.chunks[axis]) * longest_group
```

Comment:
```suggestion
    max_axis_chunks = max(x.chunks[axis])
```

Reply:
I need a bit more, but we can definitely simplify these formulas a lot